lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -4348,11 +4348,16 @@ void initPrimitives() {
     void initSerialPrimitives();
     initSerialPrimitives();
 
-#ifdef __APPLE__
+#ifdef SC_AUDIO_API_COREAUDIO
     void initCoreAudioPrimitives();
     initCoreAudioPrimitives();
 #endif
 
+#ifdef SC_AUDIO_API_PORTAUDIO
+    void initPortAudioPrimitives();
+    initPortAudioPrimitives();
+#endif
+
```

Comment:
these two paragraphs would become

```cpp
#if defined(SC_AUDIO_API_COREAUDIO) || defined(SC_AUDIO_API_PORTAUDIO)
    initAudioBackendPrimitives();
#endif
```

Reply:
> so, i'm thinking that to avoid code duplication it's probably best to merge `SC_CoreAudioPrim.cpp` and `SC_PortAudioPrim.cpp` into a new C++ file, `SC_AudioBackendPrim.cpp` (feel free to choose a better name, i'm not sure that one is accurate).

Below you also suggest refactoring what I wrote for PortAudio part. How would you like the combined file structured? Should CoreAudio part be also refactored?

Reply:
no, the CoreAudio part doesn't need to be refactored, you can just stick a ifdef around it. if i've read the code right, then prListAudioDevices and initXPrimitives won't be duplicatedlike they are now